### ISSUE TITLE
There is something wrong about the results. Loss, RMSE, MAPE all is zero.

### ISSUE THREAD

**Issue by Hevislee** [liuxu77/LargeST#7]
Log directory: ./experiments/lstm/GLA/
2024-07-24 14:05:42,361 - Namespace(device='cuda:0', dataset='GLA', years='2019', model_name='lstm', seed=2024, bs=32, seq_len=12, horizon=12, input_dim=3, output_dim=1, mode='train', max_epochs=100, patience=30, init_dim=32, hid_dim=64, end_dim=512, layer=2, lrate=0.001, wdecay=0.0001, dropout=0.1, clip_grad_value=5)
2024-07-24 14:05:55,062 - Data shape: (105120, 3834, 3)
2024-07-24 14:06:07,772 - Sample num: 63058, Batch num: 1970
2024-07-24 14:06:20,460 - Sample num: 21019, Batch num: 656
2024-07-24 14:06:33,156 - Sample num: 21020, Batch num: 656
2024-07-24 14:06:34,185 - The number of parameters: 97932
2024-07-24 14:06:34,185 - Start training!
Check mask value tensor(0)
2024-07-24 14:13:28,434 - Epoch: 001, Train Loss: 0.0000, Train RMSE: 0.0000, Train MAPE: 0.0000, Valid Loss: 0.0000, Valid RMSE: 0.0000, Valid MAPE: 0.0000, Train Time: 310.5966s/epoch, Valid Time: 103.6521s, LR: 1.0000e-03
2024-07-24 14:13:28,438 - Val loss decrease from inf to 0.0000
2024-07-24 14:20:24,661 - Epoch: 002, Train Loss: 0.0000, Train RMSE: 0.0000, Train MAPE: 0.0000, Valid Loss: 0.0000, Valid RMSE: 0.0000, Valid MAPE: 0.0000, Train Time: 312.5177s/epoch, Valid Time: 103.7053s, LR: 1.0000e-03
2024-07-24 14:27:27,512 - Epoch: 003, Train Loss: 0.0000, Train RMSE: 0.0000, Train MAPE: 0.0000, Valid Loss: 0.0000, Valid RMSE: 0.0000, Valid MAPE: 0.0000, Train Time: 317.2444s/epoch, Valid Time: 105.6064s, LR: 1.0000e-03
2024-07-24 14:34:31,354 - Epoch: 004, Train Loss: 0.0000, Train RMSE: 0.0000, Train MAPE: 0.0000, Valid Loss: 0.0000, Valid RMSE: 0.0000, Valid MAPE: 0.0000, Train Time: 320.2488s/epoch, Valid Time: 103.5923s, LR: 1.0000e-03
2024-07-24 14:41:35,001 - Epoch: 005, Train Loss: 0.0000, Train RMSE: 0.0000, Train MAPE: 0.0000, Valid Loss: 0.0000, Valid RMSE: 0.0000, Valid MAPE: 0.0000, Train Time: 317.7777s/epoch, Valid Time: 105.8682s, LR: 1.0000e-03

**Comment by liuxu77** [comment]
Hi, could you please provide more information, e.g., the environment, what you have modified.

**Comment by Hevislee** [comment]
> Hi, could you please provide more information, e.g., the environment, what you have modified.

I’ve solved the problem, because the torch version. Thanks for your excellent work.

**Comment by Hevislee** [comment]
Have solved.

**Comment by werdnabae** [comment]
Can you explain in more detail what your solution was? I'm having the same issue.

**Comment by werdnabae** [comment]
> Can you explain in more detail what your solution was? I'm having the same issue.

I solved it, the problem was caused by a missing hour in the data due to Daylight Saving Time—those timestamps didn’t actually exist. Once I removed them, the code worked as expected.
